src/UI/PaellaConfig/PaellaConfigFormBuilder.php

```diff
@@ -58,47 +63,112 @@ public function __construct(
+            ->tag($this->txt(self::F_PAELLA_PLAYER_FALLBACK_CAPTIONS_OPTION), $commonCaptions, $this->txt(self::F_PAELLA_PLAYER_FALLBACK_CAPTIONS_OPTION . '_info'))
+            ->withUserCreatedTagsAllowed(true)
+            ->withValue($defaultCaptions);
+
         return $this->ui_factory->input()->container()->form()->standard(
             $form_action,
             $inputs
         );
     }
 
-    private function getPaellaPlayerPathInput(bool $live, string $option, string $url): Input
+    private function generateSwichableGroupWithUrl(string $link, string $option, string $url, string $text, bool $required): Input
     {
         $f = $this->ui_factory->input()->field();
-        $link = $this->ui_renderer->render(
-            $this->ui_factory->link()->standard(
-                $this->plugin->txt(self::F_PAELLA_PLAYER_DEFAULT . "_link"),
-                $live ? PluginConfig::PAELLA_DEFAULT_PATH_LIVE : PluginConfig::PAELLA_DEFAULT_PATH
-            )
-        );
-
         return $f->switchableGroup([
             PluginConfig::PAELLA_OPTION_DEFAULT => $f->group([], $this->plugin->txt("pp_default_string") . " " . $link),
-            PluginConfig::PAELLA_OPTION_URL => $f->group([
-                'url' => $f->text($this->plugin->txt('link'))
-                           ->withByline($this->plugin->txt('pp_link_info'))
-                           ->withRequired(true)
-                           ->withValue($url)
-            ], $this->plugin->txt('pp_url'))
-        ], $this->txt(self::F_PAELLA_PLAYER_OPTION))
-                 ->withValue($option)
-                 ->withRequired(true);
+            PluginConfig::PAELLA_OPTION_URL => $f->group(
+                [
+                    'url' => $f->text($this->plugin->txt('link'))
+                                ->withByline($this->plugin->txt('pp_link_info'))
+                                ->withRequired(true)
+                                ->withValue($url)
+                ],
+                $this->plugin->txt('pp_url')
+            )
+        ], $this->txt($text))
+            ->withByline($this->txt($text . '_info'))
+            ->withValue($option)
+            ->withRequired($required);
+    }
+
+    private function buildInlineDownload(string $file_id): string
+    {
+        if (!$file_id || !$this->paellaStorageService->exists($file_id)) {
+            return '';
+        }
+        $fileAsBase64 = $this->paellaStorageService->getFileAsBase64($file_id);
+        $fileInfo = $this->paellaStorageService->getFileInfo($file_id);
+        return '<a href="data:text/vtt;base64,'
+            . $fileAsBase64
+            . '" target="blank" download="' . $fileInfo['name'] . '">Download</a>';
     }
 
     private function txt(string $string): string
     {
         return $this->plugin->txt('config_' . $string);
     }
+
+    private function getAvailablePlayerLanguages(): array
```

Comment:
It only happens in admin settings, and the reason behind it was to provide the further expansion of the languages automatically by only adding a json file in that directory.

So, in terms of occurrence, we are safe to keep this, because as I said, only happens one time in admin settings when visiting settings player tab!